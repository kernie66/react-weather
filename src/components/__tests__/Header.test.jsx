import {
  render,
  renderWithNotifications,
  screen,
  userEvent,
  waitForElementToBeRemoved,
} from '../../../testing-utils';
import { testQueryClient } from '../../../testing-utils/render.jsx';
import { expect } from 'vitest';
import Header from '../Header.jsx';
import { locationHistoryData } from './locationHistoryData.js';
import { defaultAddress } from '../../atoms/locationStates.js';

const fakeLocationHistory = locationHistoryData;
const fakeLocationOption =
  fakeLocationHistory[fakeLocationHistory.length - 3].address;

describe('test Header and history selection', () => {
  beforeEach(() => {
    testQueryClient.removeQueries();
    localStorage.clear();
  });

  it('renders the header with buttons and empty history selector', async () => {
    const user = userEvent.setup();
    const snapshot = render(<Header />);

    const historyButton = await screen.findByRole('button', {
      name: /historik/i,
    });
    expect(historyButton).toBeInTheDocument();
    expect(
      await screen.findByRole('button', { name: /meny/i })
    ).toBeInTheDocument();
    expect(
      await screen.findByRole('button', { name: /öppna karta/i })
    ).toBeInTheDocument();
    expect(screen.getByText(defaultAddress)).toBeInTheDocument();
    expect(snapshot).toMatchSnapshot();

    // Click location text button
    await user.click(historyButton);
    expect(await screen.findByRole('dialog')).toBeInTheDocument();
    const noHistoryOption = await screen.findByRole('option', {
      name: /ingen historik/i,
    });
    expect(noHistoryOption).toBeInTheDocument();

    // Click on no history option to close dialog
    await user.click(noHistoryOption);
    await waitForElementToBeRemoved(() =>
      screen.queryByRole('dialog')
    );
    expect(screen.getByText(defaultAddress)).toBeInTheDocument();
  });

  it('renders the header with buttons and selects history location with mouse', async () => {
    const user = userEvent.setup();
    localStorage.setItem(
      'locationHistory',
      JSON.stringify(fakeLocationHistory)
    );

    renderWithNotifications(<Header />);

    const historyButton = await screen.findByRole('button', {
      name: /historik/i,
    });
    expect(historyButton).toBeInTheDocument();
    expect(screen.getByText(defaultAddress)).toBeInTheDocument();
    expect(screen.queryByRole('listbox')).toBeNull();

    // Click location name button
    await user.click(historyButton);
    expect(await screen.findByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('listbox')).toBeInTheDocument();
    expect(screen.getAllByRole('option')).toHaveLength(
      fakeLocationHistory.length
    );

    // Select third option to update location and close dialog
    const thirdOption = await screen.findByRole('option', {
      name: fakeLocationOption,
    });
    expect(thirdOption).toBeInTheDocument();
    await user.click(thirdOption);
    await waitForElementToBeRemoved(() =>
      screen.queryByRole('dialog')
    );

    // Expect location in both history button and in notification
    expect(screen.getAllByText(fakeLocationOption)).toHaveLength(2);
    expect(screen.getByRole('alert')).toBeInTheDocument();
    expect(
      screen.getByText(/väderposition uppdaterad/i)
    ).toBeInTheDocument();

    // Close notification
    const closeNotification = screen.getByLabelText(/stäng notis/i);
    expect(closeNotification).toBeInTheDocument();
    await user.click(closeNotification);
    await waitForElementToBeRemoved(() =>
      screen.queryByRole('alert')
    );

    // Click location name button again
    await user.click(historyButton);
    expect(await screen.findByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('listbox')).toBeInTheDocument();

    // Click outside to close dialog without updating location
    await user.click(document.body);
    await waitForElementToBeRemoved(() =>
      screen.queryByRole('dialog')
    );
    expect(screen.getByText(fakeLocationOption)).toBeInTheDocument();
  });

  it('renders the header with buttons and selects history location with keyboard', async () => {
    const user = userEvent.setup();
    localStorage.setItem(
      'locationHistory',
      JSON.stringify(fakeLocationHistory)
    );
    console.log('Starting third test.....');
    render(<Header />);

    const historyButton = await screen.findByRole('button', {
      name: /historik/i,
    });
    expect(historyButton).toBeInTheDocument();
    expect(screen.getByText(defaultAddress)).toBeInTheDocument();
    expect(screen.queryByRole('listbox')).toBeNull();

    screen.debug(undefined, Infinity);
    // Click location name button
    await user.click(historyButton);
    expect(await screen.findByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('listbox')).toBeInTheDocument();
    expect(screen.getAllByRole('option')).toHaveLength(
      fakeLocationHistory.length
    );

    // Select third option to update location and close dialog
    const thirdOption = await screen.findByRole('option', {
      name: fakeLocationOption,
    });
    expect(thirdOption).toBeInTheDocument();
    await user.click(thirdOption);
    await waitForElementToBeRemoved(() =>
      screen.queryByRole('dialog')
    );
    expect(screen.getByText(fakeLocationOption)).toBeInTheDocument();

    // Click location name button again
    await user.click(historyButton);
    expect(await screen.findByRole('dialog')).toBeInTheDocument();
    expect(screen.getByRole('listbox')).toBeInTheDocument();

    // Click outside to close dialog without updating location
    await user.click(document.body);
    await waitForElementToBeRemoved(() =>
      screen.queryByRole('dialog')
    );
    expect(screen.getByText(fakeLocationOption)).toBeInTheDocument();
  });
});
