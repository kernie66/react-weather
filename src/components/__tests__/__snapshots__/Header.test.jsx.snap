// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test Header > renders the header with buttons and empty history selector 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        class="weather-header m_4081bf90 mantine-Group-root"
        style="--group-gap: 0rem; --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
      >
        <button
          aria-controls=""
          aria-expanded="false"
          aria-haspopup="menu"
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="subtle"
          id=""
          style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6l16 0"
                />
                <path
                  d="M4 12l16 0"
                />
                <path
                  d="M4 18l16 0"
                />
              </svg>
              <span
                class="m_515a97f8 mantine-VisuallyHidden-root"
              >
                Meny
              </span>
            </span>
          </span>
        </button>
        <div
          class=""
          style="padding-inline: calc(0.25rem * var(--mantine-scale));"
        >
          <div
            class="m_4451eb3a mantine-Center-root"
          >
            <p
              class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
              xs="10"
            >
              <span
                class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
                style="padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(13.75rem * var(--mantine-scale));"
              >
                Väderstation:
              </span>
              <button
                aria-controls=""
                aria-expanded="false"
                aria-haspopup="dialog"
                class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
                data-size="lg"
                data-variant="subtle"
                data-with-right-section="true"
                id=""
                justify="space-between"
                style="--button-justify: space-between; --button-height: var(--button-height-lg); --button-padding-x: var(--button-padding-x-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; vertical-align: baseline; padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(41.5rem * var(--mantine-scale));"
                type="button"
              >
                <span
                  class="m_80f1301b mantine-Button-inner"
                >
                  <span
                    class="m_811560b9 mantine-Button-label"
                  >
                    <span
                      class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
                      data-line-clamp="true"
                      style="--text-line-clamp: 1; text-align: left; max-width: calc(41rem * var(--mantine-scale));"
                    >
                      Rotebro, Sollentuna, Sverige
                    </span>
                    <span
                      class="m_515a97f8 mantine-VisuallyHidden-root"
                    >
                      Historik
                    </span>
                  </span>
                  <span
                    class="m_a74036a mantine-Button-section"
                    data-position="right"
                  >
                    <span />
                  </span>
                </span>
              </button>
            </p>
          </div>
        </div>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="subtle"
          style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem; color: crimson;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 18.5l-3 -1.5l-6 3v-13l6 -3l6 3l6 -3v7.5"
                />
                <path
                  d="M9 4v13"
                />
                <path
                  d="M15 7v5.5"
                />
                <path
                  d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z"
                />
                <path
                  d="M19 18v.01"
                />
              </svg>
              <span
                class="m_515a97f8 mantine-VisuallyHidden-root"
              >
                Öppna karta
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div>
        <div
          class="m_9df02822 mantine-Modal-root"
          data-full-screen="true"
          style="--modal-y-offset: 5dvh; --mb-z-index: 200;"
        />
      </div>
    </div>
    <div
      data-portal="true"
    />
    <div
      data-portal="true"
    />
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      class="weather-header m_4081bf90 mantine-Group-root"
      style="--group-gap: 0rem; --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
    >
      <button
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="menu"
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="subtle"
        id=""
        style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6l16 0"
              />
              <path
                d="M4 12l16 0"
              />
              <path
                d="M4 18l16 0"
              />
            </svg>
            <span
              class="m_515a97f8 mantine-VisuallyHidden-root"
            >
              Meny
            </span>
          </span>
        </span>
      </button>
      <div
        class=""
        style="padding-inline: calc(0.25rem * var(--mantine-scale));"
      >
        <div
          class="m_4451eb3a mantine-Center-root"
        >
          <p
            class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
            xs="10"
          >
            <span
              class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
              style="padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(13.75rem * var(--mantine-scale));"
            >
              Väderstation:
            </span>
            <button
              aria-controls=""
              aria-expanded="false"
              aria-haspopup="dialog"
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-size="lg"
              data-variant="subtle"
              data-with-right-section="true"
              id=""
              justify="space-between"
              style="--button-justify: space-between; --button-height: var(--button-height-lg); --button-padding-x: var(--button-padding-x-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; vertical-align: baseline; padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(41.5rem * var(--mantine-scale));"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  <span
                    class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
                    data-line-clamp="true"
                    style="--text-line-clamp: 1; text-align: left; max-width: calc(41rem * var(--mantine-scale));"
                  >
                    Rotebro, Sollentuna, Sverige
                  </span>
                  <span
                    class="m_515a97f8 mantine-VisuallyHidden-root"
                  >
                    Historik
                  </span>
                </span>
                <span
                  class="m_a74036a mantine-Button-section"
                  data-position="right"
                >
                  <span />
                </span>
              </span>
            </button>
          </p>
        </div>
      </div>
      <button
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="subtle"
        style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem; color: crimson;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 18.5l-3 -1.5l-6 3v-13l6 -3l6 3l6 -3v7.5"
              />
              <path
                d="M9 4v13"
              />
              <path
                d="M15 7v5.5"
              />
              <path
                d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z"
              />
              <path
                d="M19 18v.01"
              />
            </svg>
            <span
              class="m_515a97f8 mantine-VisuallyHidden-root"
            >
              Öppna karta
            </span>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test Header and history selection > renders the header with buttons and empty history selector 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <div
        class="weather-header m_4081bf90 mantine-Group-root"
        style="--group-gap: 0rem; --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
      >
        <button
          aria-controls=""
          aria-expanded="false"
          aria-haspopup="menu"
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="subtle"
          id=""
          style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6l16 0"
                />
                <path
                  d="M4 12l16 0"
                />
                <path
                  d="M4 18l16 0"
                />
              </svg>
              <span
                class="m_515a97f8 mantine-VisuallyHidden-root"
              >
                Meny
              </span>
            </span>
          </span>
        </button>
        <div
          class=""
          style="padding-inline: calc(0.25rem * var(--mantine-scale));"
        >
          <div
            class="m_4451eb3a mantine-Center-root"
          >
            <p
              class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
              xs="10"
            >
              <span
                class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
                style="padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(13.75rem * var(--mantine-scale));"
              >
                Väderstation:
              </span>
              <button
                aria-controls=""
                aria-expanded="false"
                aria-haspopup="dialog"
                class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
                data-size="lg"
                data-variant="subtle"
                data-with-right-section="true"
                id=""
                justify="space-between"
                style="--button-justify: space-between; --button-height: var(--button-height-lg); --button-padding-x: var(--button-padding-x-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; vertical-align: baseline; padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(41.5rem * var(--mantine-scale));"
                type="button"
              >
                <span
                  class="m_80f1301b mantine-Button-inner"
                >
                  <span
                    class="m_811560b9 mantine-Button-label"
                  >
                    <span
                      class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
                      data-line-clamp="true"
                      style="--text-line-clamp: 1; text-align: left; max-width: calc(41rem * var(--mantine-scale));"
                    >
                      Rotebro, Sollentuna, Sverige
                    </span>
                    <span
                      class="m_515a97f8 mantine-VisuallyHidden-root"
                    >
                      Historik
                    </span>
                  </span>
                  <span
                    class="m_a74036a mantine-Button-section"
                    data-position="right"
                  >
                    <span />
                  </span>
                </span>
              </button>
            </p>
          </div>
        </div>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="subtle"
          style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem; color: crimson;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 18.5l-3 -1.5l-6 3v-13l6 -3l6 3l6 -3v7.5"
                />
                <path
                  d="M9 4v13"
                />
                <path
                  d="M15 7v5.5"
                />
                <path
                  d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z"
                />
                <path
                  d="M19 18v.01"
                />
              </svg>
              <span
                class="m_515a97f8 mantine-VisuallyHidden-root"
              >
                Öppna karta
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      data-portal="true"
    >
      <div>
        <div
          class="m_9df02822 mantine-Modal-root"
          data-full-screen="true"
          style="--modal-y-offset: 5dvh; --mb-z-index: 200;"
        />
      </div>
    </div>
    <div
      data-portal="true"
    />
    <div
      data-portal="true"
    />
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <div
      class="weather-header m_4081bf90 mantine-Group-root"
      style="--group-gap: 0rem; --group-align: center; --group-justify: space-between; --group-wrap: wrap;"
    >
      <button
        aria-controls=""
        aria-expanded="false"
        aria-haspopup="menu"
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="subtle"
        id=""
        style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6l16 0"
              />
              <path
                d="M4 12l16 0"
              />
              <path
                d="M4 18l16 0"
              />
            </svg>
            <span
              class="m_515a97f8 mantine-VisuallyHidden-root"
            >
              Meny
            </span>
          </span>
        </span>
      </button>
      <div
        class=""
        style="padding-inline: calc(0.25rem * var(--mantine-scale));"
      >
        <div
          class="m_4451eb3a mantine-Center-root"
        >
          <p
            class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
            xs="10"
          >
            <span
              class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
              style="padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(13.75rem * var(--mantine-scale));"
            >
              Väderstation:
            </span>
            <button
              aria-controls=""
              aria-expanded="false"
              aria-haspopup="dialog"
              class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
              data-size="lg"
              data-variant="subtle"
              data-with-right-section="true"
              id=""
              justify="space-between"
              style="--button-justify: space-between; --button-height: var(--button-height-lg); --button-padding-x: var(--button-padding-x-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; vertical-align: baseline; padding-inline: calc(0.25rem * var(--mantine-scale)); max-width: calc(41.5rem * var(--mantine-scale));"
              type="button"
            >
              <span
                class="m_80f1301b mantine-Button-inner"
              >
                <span
                  class="m_811560b9 mantine-Button-label"
                >
                  <span
                    class="mantine-focus-auto _outlineHeader_597f95 m_b6d8b162 mantine-Text-root"
                    data-line-clamp="true"
                    style="--text-line-clamp: 1; text-align: left; max-width: calc(41rem * var(--mantine-scale));"
                  >
                    Rotebro, Sollentuna, Sverige
                  </span>
                  <span
                    class="m_515a97f8 mantine-VisuallyHidden-root"
                  >
                    Historik
                  </span>
                </span>
                <span
                  class="m_a74036a mantine-Button-section"
                  data-position="right"
                >
                  <span />
                </span>
              </span>
            </button>
          </p>
        </div>
      </div>
      <button
        class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
        data-variant="subtle"
        style="--button-bg: transparent; --button-hover: var(--mantine-color-blue-light-hover); --button-color: var(--mantine-color-blue-light-color); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; padding-inline: 0rem; color: crimson;"
        type="button"
      >
        <span
          class="m_80f1301b mantine-Button-inner"
        >
          <span
            class="m_811560b9 mantine-Button-label"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 18.5l-3 -1.5l-6 3v-13l6 -3l6 3l6 -3v7.5"
              />
              <path
                d="M9 4v13"
              />
              <path
                d="M15 7v5.5"
              />
              <path
                d="M21.121 20.121a3 3 0 1 0 -4.242 0c.418 .419 1.125 1.045 2.121 1.879c1.051 -.89 1.759 -1.516 2.121 -1.879z"
              />
              <path
                d="M19 18v.01"
              />
            </svg>
            <span
              class="m_515a97f8 mantine-VisuallyHidden-root"
            >
              Öppna karta
            </span>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
